lib/main/STM32H7/Drivers/STM32H7xx_HAL_Driver/Inc/stm32h7xx_ll_spi.h

```diff
@@ -17,6 +17,9 @@
   ******************************************************************************
   */
 
+// Avoid strict aliasing warnings from deferencing type-punned pointers
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
```

Comment:
@jflyper: Setting this pragma here in an unbounded way is risky - especially since this is a header file, and consequently the warning will be ignored in all files including this header.

Can you please turn this into a bounded pragma (see https://github.com/betaflight/betaflight/blob/f1cd99e6b61f276e581a5856192dc4fb1241125c/src/main/fc/rc.c#L627-L632)?